src/main/java/com/labsynch/labseer/service/SaltServiceImpl.java

```diff
@@ -99,4 +147,207 @@ private void saveSalt(CmpdRegMolecule mol) throws IOException, CmpdRegMolFormatE
+				currSalt.setProperty("charge", String.valueOf(salt.getCharge()));
+				sdfWriter.writeMol(currSalt);
+			} catch (CmpdRegMolFormatException e) {
+				logger.error("bad structure error: " + salt.getMolStructure());
+			} catch (IOException e) {
+				logger.error("IO error reading in molfile");
+				e.printStackTrace();
+			}
+
+		}
+		try {
+			sdfWriter.close();
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+		logger.debug("ou	tput SDF: " + sdfWriter.getBufferString());
+		return sdfWriter.getBufferString();
+	}
+
+	// Helper Method Used to Check Dependencies Within ACAS
+	private Map<String, HashSet<String>> checkACASDependencies(
+		Map<String, HashSet<String>> acasDependencies) throws MalformedURLException, IOException {
+			String url = propertiesUtilService.getAcasURL() + "compounds/checkBatchDependencies";
+			BatchCodeDependencyDTO request = new BatchCodeDependencyDTO(acasDependencies.keySet());
+			String json = request.toJson();
+			String responseJson = SimpleUtil.postRequestToExternalServer(url, json, null);
+			BatchCodeDependencyDTO responseDTO = BatchCodeDependencyDTO.fromJsonToBatchCodeDependencyDTO(responseJson);
+			if (responseDTO.getLinkedDataExists()) {
+				for (CodeTableDTO experimentCodeTable : responseDTO.getLinkedExperiments()) {
+					String experimentCodeAndName = experimentCodeTable.getCode() + " ( " + experimentCodeTable.getName()
+							+ " )";
+					if (acasDependencies.containsKey(experimentCodeTable.getComments())) {
+						acasDependencies.get(experimentCodeTable.getComments()).add(experimentCodeAndName);
+					} else {
+						HashSet<String> codes = new HashSet<String>();
+						codes.add(experimentCodeAndName);
+						acasDependencies.put(experimentCodeTable.getComments(), codes);
+					}
+				}
+			}
+			return acasDependencies;
+		}
+
+	// Helper Method to Check Dependent Containers; Utilized in Method Below It
+	private Collection<ContainerBatchCodeDTO> checkDependentACASContainers(Set<String> batchCodes)
+		throws MalformedURLException, IOException {
+			String url = propertiesUtilService.getAcasURL() + "containers/getContainerDTOsByBatchCodes";
+			String json = (new JSONSerializer()).serialize(batchCodes);
+			String responseJson = SimpleUtil.postRequestToExternalServer(url, json, null);
+			Collection<ContainerBatchCodeDTO> responseDTOs = ContainerBatchCodeDTO
+					.fromJsonArrayToContainerBatchCoes(responseJson);
+			return responseDTOs;
+		}
+
+	// Method to Check for Dependent Data 
+	public PurgeSaltDependencyCheckResponseDTO checkDependentData(Salt salt)
+	{
+		Map<String, HashSet<String>> acasDependencies = new HashMap<String, HashSet<String>>();
+		Map<String, HashSet<String>> cmpdRegDependencies = new HashMap<String, HashSet<String>>();
+		HashSet<String> dependentSingleRegLots = new HashSet<String>();
+		int numberOfParents = 0;
+		int numberOfSaltForms = 0;
+		int numberOfLots = 0;
+		Collection<Parent> parents = Parent.findParentsByCdId(salt.getCdId()).getResultList();
```

Comment:
This line seems to be looking up Parent compounds whose structure ID matches the structure ID of the salt. This isn't correct.

To find dependencies for Salts, we'll need to find any SaltForms that contain this Salt, then find the dependencies of those SaltForms. We should chat about this.

Reply:
Ah makes sense. Happy to chat about this further. I need to gain a better understanding of identifiers and object relationships in ACAS.